src/test/scala/org/ergoplatform/nodeView/viewholder/ErgoNodeViewHolderSpec.scala

```diff
@@ -67,6 +67,24 @@ class ErgoNodeViewHolderSpec extends ErgoPropertyTest with HistoryTestHelpers wi
     getBestHeaderOpt shouldBe Some(block.header)
   }
 
+  private val t3a = TestCase("do not apply block headers in invalid order") { fixture =>
+    import fixture._
+    val (us, bh) = createUtxoState(parameters)
+    val parentBlock = validFullBlock(None, us, bh)
+    val block = validFullBlock(Some(parentBlock), us, bh)
+
+    getBestHeaderOpt shouldBe None
+    getHistoryHeight shouldBe ErgoHistory.EmptyHistoryHeight
+
+    subscribeEvents(classOf[SyntacticallySuccessfulModifier])
+
+    //sending header
+    nodeViewHolderRef ! ModifiersFromRemote(List(block.header))
+    expectNoMsg()
+
+    getHistoryHeight shouldBe 0
```

Comment:
I guess now you need to apply parentBlock and then block and be sure that both are applied